Core/src/Detector/detail/CuboidalDetectorHelper.cpp

```diff
@@ -190,9 +191,7 @@ Acts::Experimental::detail::CuboidalDetectorHelper::connect(
 
     for (auto [is, index] : enumerate(portalSets[toUnderlying(mergeValue)])) {
       // Check if you need to skip due to selections
-      if (!selectedOnly.empty() &&
-          std::find(selectedOnly.begin(), selectedOnly.end(), index) ==
-              selectedOnly.end()) {
+      if (!selectedOnly.empty() && !rangeContainsValue(selectedOnly, index)) {
```

Comment:
Isn't this a tautology?

Reply:
I don't think so. This avoids that we continue for empty vector. In case of an empty vector we have:
`vec.empty() == true`
`rangeContainsValue(vec, element) == false`.

Reply:
By case analysis:

* Empty vector: expression resolves to `!true && !false` is `false` is the same as `!rangeContainsValue`
* Non-empty vector not containing the element: expression resolves to `!false && !false` is `true` is the same as `!rangeContainsValue`
* Non-empty vector containing the element: expression resolves to `!false && !true` is `false` is the same as `!rangeContainsValue`

Right?

Reply:
Wow yes, I don't know how I missed that; you are right!